auditbeat/module/audit/file/eventreader_fsnotify.go

```diff
@@ -0,0 +1,119 @@
+// +build linux freebsd openbsd netbsd windows darwin
+
+package file
+
+import (
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/fsnotify/fsnotify"
+)
+
+// NewEventReader creates a new EventReader backed by fsnotify.
+func NewEventReader(c Config) (EventReader, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	return &reader{watcher, c, make(chan Event, 1), make(chan error, 1)}, nil
+}
+
+type reader struct {
+	watcher *fsnotify.Watcher
+	config  Config
+	outC    chan Event
+	errC    chan error
+}
+
+func (r *reader) Start(done <-chan struct{}) (<-chan Event, error) {
+	var paths []string
+	for _, filePaths := range r.config.Paths {
+		paths = append(paths, filePaths...)
+	}
+
+	for _, p := range paths {
+		if err := r.watcher.Add(p); err != nil {
```

Comment:
Every time I see this I'm thinking if we should add a new prospector type in filebeat based on inotify which only works with "most" OS. Could make to code a lot simpler.